api/src/main/java/com/datastrato/gravitino/NameIdentifier.java

```diff
@@ -220,4 +228,33 @@ public static void check(boolean condition, String message, Object... args) {
       throw new IllegalNameIdentifierException(String.format(message, args));
     }
   }
+
+  public NameIdentifier child(String childName) {
+    String[] levels = ArrayUtils.add(namespace.levels(), this.name);
+    Namespace ns = Namespace.of(levels);
+    return NameIdentifier.of(ns, childName);
+  }
+
+  public boolean hasParent() {
+    return this != ROOT;
+  }
+
+  public NameIdentifier parent() {
+    if (this == ROOT) {
+      throw new IllegalArgumentException("Cannot get parent of ROOT");
+    }
+
+    String[] levels = namespace.levels();
+    if (levels.length == 0) {
+      return ROOT;
+    }
+
+    Namespace ns = Namespace.of(ArrayUtils.subarray(levels, 0, levels.length - 1));
+    return NameIdentifier.of(ns, levels[levels.length - 1]);
+  }
+
+  public String[] levels() {
+    String[] levels = namespace.levels();
+    return ArrayUtils.add(levels, this.name);
+  }
```

Comment:
We should carefully add the interfaces as it will be used by the users? If this is just for use internally, we can add some helper methods separately.

Reply:
Got it.